### DISCUSSION TITLE
Install my MAUI App in another computer

### DISCUSSION THREAD

**Discussion by ahipki** [dotnet/maui#23560] [Q&A] [answered]
Hello,

I created a **package** of my MAUI App.
I **installed** it on my **dev computer** (create new project MAUI > Publish).
It **works** on this computer.
So I zipped my package :
net8.0-windows10.0.19041.0/win10-x64/AppPackages
I downloaded and unzipped it on a new computer.
I **installed** it, it launched and told "You must install .Net Desktop Runtime to run this application" so I clicked on Download it now. 
So on my **new computer** I have Microsoft Windows Desktop Runtime - 8.0.6 (x64) on my apps list.
And I launched it again, **nothing happens**.
Should I install another microsoft thing on this new computer ? .Net 8.0 ? Microsoft .Net Runtime ? SDK something ? Visual C++ ? (So many Microsoft apps on my dev computer)

Thanks for your help.
I watched and read tutorials but it's tutorials to create an MAUI App then deploy it on the same machine...

(sorry, English is not my native language)

**Comment by vitek-karas** [comment]
I assume by "nothing happens" you mean you don't get an error anymore, but the app doesn't start either.
Could you please look into event log? (Search for "event viewer" in start menu) There might be an application error about the failure to start.

**Reply by ahipki** [reply]
Ah thanks : I had 0 event but it doesn't refresh automatically so I didn't see last time.
So the error for myApp (I don't have this error on a basic app) comes from "C:/Program Files/WindowsApps/microsoft.windowsappruntile.1.4_4000.1227.13637.0_x64_..../Microsoft.ui.xaml.dll 
It doesn't help me :/ I don't know if I need to install something about microsoft on the computer or change some nugets in my app...

I tried installing Windows App SDK 1.5 (the last one) and re install the app, the same error. I also changed the microsoft Windows Desktop Runtime 8.0.6 for 8.0.7 and same event error.

**Reply by vitek-karas** [reply]
What is the exact error you're seeing?

**Reply by ahipki** [reply]
The same error than https://answers.microsoft.com/en-us/windows/forum/all/windowsuixamldll-crash-at-startup/c4ec97ed-2ebe-4ef8-8505-23c1f6faeda7 
![image](https://github.com/dotnet/maui/assets/9572556/617db86b-6476-4a98-93c0-87eedee2886e)

**Reply by ahipki** [reply]
I just tried this solution https://answers.microsoft.com/en-us/windows/forum/all/how-to-fix-windowsuixamldll/6c63de66-1191-4314-abe8-81988e9fba97 but same problem

**Reply by vitek-karas** [reply]
Sorry - this is outside of what I know. Maybe somebody here would know who to contact on the Windows UI side as it seems to be a problem in that area.

**Reply by ahipki** [reply]
Thanks for your time. It was helpful for me !

**Comment by daltzctr** [comment]
MAUI applications by default are shipped as PACKAGED applications. PACKAGED applications cannot be ran with just the exe. It must be bootstrapped into the system.

If you want to distribute a packaged app, the only real way to do this is through the Microsoft Store. You can alternatively setup a local signing cert, publish and local sign, then hand off an install script to install.

If you want a traditional exe install, you must do UNPACKAGED. You can read up on UNPACKAGED apps [here](https://learn.microsoft.com/en-us/dotnet/maui/windows/deployment/publish-unpackaged-cli?view=net-maui-8.0)

**Reply by ahipki** [reply]
I tried with an unpackaged folder and when I launched with the exe, I finally have the same event Error : Microsoft.ui.xaml.dll V3.1.4 has a problem

**Reply by daltzctr** [reply]
Do you actually have the Windows App SDK installed on your client's computer? This is required or you need to distribute as `WindowsAppSDKSelfContained`

**Reply by ahipki** [reply, marked as answer]
Yes I tried everything on the Microsoft  doc :) 
But i found this topic : https://github.com/dotnet/maui/issues/9839 and I will try to find the not catched exceptions, maybe it will work after that

**Reply by ahipki** [reply]
I found the exception
I had 
string settingsPath = Path.Combine(ProjectSourcePath.Value, "Properties");
and I changed it for : string settingsPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Properties");
And now it's ok.

Thanks all for your time
